src/ai/backend/common/clients/valkey_client/client.py

```diff
@@ -455,3 +463,103 @@ async def wrapper(*args, **kwargs) -> R:
         return decorator
 
     return valkey_decorator
+
+
+def create_layer_aware_valkey_decorator_with_default(
+    layer: LayerType,
+):
+    """
+    Factory function to create layer-aware valkey decorators.
+
+    Args:
+        layer: The layer type for metric observation
+
+    Returns:
+        A valkey_decorator function configured for the specified layer
+    """
+
+    def valkey_decorator(
+        *,
+        retry_count: int = 1,
+        retry_delay: float = 0.1,
+        default_return: Union[R, _Sentinel] = _SENTINEL,
+    ):
```

Comment:
It seems better to handle `default_return` as a required parameter instead of needing to declare a SENTINEL.
Since it's a newly created decorator anyway, there's no need to consider existing compatibility.